src/main/groovy/io/github/cosmicsilence/scalafix/ScalaSourceSet.groovy

```diff
@@ -38,22 +44,34 @@ class ScalaSourceSet {
     }
 
     ScalaCompile getCompileTask() {
-        return project.tasks.getByName(getCompileTaskName(sourceSet))
+        return compileTask
     }
 
     @Memoized
     Optional<String> getScalaVersion() {
         def scalaRuntime = project.extensions.findByType(ScalaRuntime)
-        def scalaJar = scalaRuntime?.findScalaJar(getCompileTask().classpath, 'library')
+        def scalaJar = scalaRuntime?.findScalaJar(compileTask.classpath, 'library')
         return Optional.ofNullable(scalaJar ? scalaRuntime.getScalaVersion(scalaJar) : null)
     }
 
     List<String> getCompilerOptions() {
-        return getCompileTask().scalaCompileOptions.additionalParameters ?: []
+        return compileTask.scalaCompileOptions.additionalParameters ?: []
     }
 
     void addCompilerOptions(List<String> opts) {
-        getCompileTask().scalaCompileOptions.additionalParameters = getCompilerOptions() + opts
+        compileTask.scalaCompileOptions.additionalParameters = getCompilerOptions() + opts
+    }
+
+    void addCompilerPlugin(String pluginCoordinates) {
+        def dependency = project.dependencies.create(pluginCoordinates)
+        def configuration = project.configurations.detachedConfiguration(dependency).setTransitive(false)
```

Comment:
Not using Gradle's own `scalaCompilerPlugins` [configuration](https://github.com/gradle/gradle/blob/master/platforms/jvm/scala/src/main/java/org/gradle/api/plugins/scala/ScalaBasePlugin.java#L95) to obtain the dependency because it might be possible that different source sets in the same Gradle project use different Scala versions. The Semanticdb plugin encodes even the patch version/segment in the artifact coordinates (https://mvnrepository.com/artifact/org.scalameta/semanticdb-scalac)

Reply:
The assumption here is that the `@Classpath` annotation on the [ScalaCompile. scalaCompilerPlugins](https://github.com/gradle/gradle/blob/8ec670c5eeddcdaba6ee840360cb4eeda788558e/platforms/jvm/scala/src/main/java/org/gradle/api/tasks/scala/ScalaCompile.java#L58C27-L58C44) field will take care of normalizing the jar input. @aishfenton / @DanielThomas please let me know if this is not correct (I couldn't test this)